File: evals/workflow_value/results/2026-09-25/cells/04/submitted-project/src/thermal_cpp_monitor/test/test_conversion.cpp
#include <gtest/gtest.h>

#include <cmath>

#include "thermal_cpp_monitor/conversion.hpp"

TEST(Conversion, ValidReadingReturnsDegreesCelsius)
{
  thermal_interfaces::msg::ThermalReading msg;
  msg.temperature_c = 21.5;
  msg.valid = true;
  EXPECT_DOUBLE_EQ(thermal_cpp_monitor::normalize(msg), 21.5);
}

TEST(Conversion, InvalidReadingReturnsNaN)
{
  thermal_interfaces::msg::ThermalReading msg;
  msg.temperature_c = 21.5;
  msg.valid = false;
  EXPECT_TRUE(std::isnan(thermal_cpp_monitor::normalize(msg)));
}

TEST(Conversion, DefaultReadingIsInvalid)
{
  thermal_interfaces::msg::ThermalReading msg;
  EXPECT_TRUE(std::isnan(thermal_cpp_monitor::normalize(msg)));
}

File: evals/workflow_value/results/2026-09-25/cells/03/submitted-project/src/thermal_py_monitor/test/test_conversion.py
import math

from thermal_interfaces.msg import ThermalReading

from thermal_py_monitor.conversion import normalize


def test_valid_reading_returns_degrees_celsius():
    msg = ThermalReading()
    msg.temperature_c = 21.5
    msg.valid = True
    assert normalize(msg) == 21.5


def test_invalid_reading_returns_nan():
    msg = ThermalReading()
    msg.temperature_c = 21.5
    msg.valid = False
    assert math.isnan(normalize(msg))
